video/profiles.go

```diff
@@ -103,22 +106,27 @@ func SetTranscodeProfiles(inputVideoStats InputVideo, reqTranscodeProfiles []Enc
 		// a single profile that matches the input video's specs with the target bitrate
 	} else if len(reqTranscodeProfiles) == 1 {
 		if reqTranscodeProfiles[0].Width == 0 && reqTranscodeProfiles[0].Height == 0 && reqTranscodeProfiles[0].Bitrate != 0 {
-			transcodeProfiles = GenerateSingleProfileWithTargetBitrate(videoTrack, reqTranscodeProfiles[0].Bitrate)
+			transcodeProfiles = GenerateSingleProfileWithTargetParams(videoTrack, reqTranscodeProfiles[0])
 			return transcodeProfiles, nil
 		}
 	}
 	return reqTranscodeProfiles, nil
 }
 
-func GenerateSingleProfileWithTargetBitrate(videoTrack InputTrack, videoBitrate int64) []EncodedProfile {
+func GenerateSingleProfileWithTargetParams(videoTrack InputTrack, videoProfile EncodedProfile) []EncodedProfile {
 	profiles := make([]EncodedProfile, 0)
+	var CRF uint = defaultCRF
+	if videoProfile.CRF != 0 {
+		CRF = videoProfile.CRF
```

Comment:
CRF = 0 is still a valid value (meaning lossless). Are we explicitly saying crf = 0 should never be allowed or should this check for values between 0 and 51?

Reply:
This is a good point. The problem with `0` is that everywhere in go and protobuf, it's the default => so, the assumption is that `0` is a not set value. 

Anyway, since as you know, there is no direct correspondence of the the `quality` value and the `crf` / `cq`, let's maybe not think about this corner case too much.